packages/core/src/stateUtils.ts

```diff
@@ -216,52 +203,24 @@ export function getAdjList<
   return adjList;
 }
 
-export function getStateValue<
-  TContext extends MachineContext,
-  TE extends EventObject
->(
-  rootNode: StateNode<TContext, TE>,
-  configuration: Configuration<TContext, TE>
+export function getStateValue(
+  rootNode: AnyStateNode,
+  configuration: AnyConfiguration
 ): StateValue {
   const config = getConfiguration(configuration);
   return getValueFromAdj(rootNode, getAdjList(config));
 }
 
-export function has<T>(iterable: Iterable<T>, item: T): boolean {
-  if (Array.isArray(iterable)) {
-    return iterable.some((member) => member === item);
-  }
-
-  if (iterable instanceof Set) {
-    return iterable.has(item);
-  }
-
-  return false; // TODO: fix
-}
-
-export function nextEvents<
-  TContext extends MachineContext,
-  TE extends EventObject
->(configuration: Array<StateNode<TContext, TE>>): Array<TE['type']> {
-  return [...new Set(flatten([...configuration.map((sn) => sn.ownEvents)]))];
-}
-
-export function isInFinalState<
-  TContext extends MachineContext,
-  TE extends EventObject
->(
-  configuration: Array<StateNode<TContext, TE>>,
-  stateNode: StateNode<TContext, TE> = configuration[0].machine.root
+export function isInFinalState(
+  configuration: Array<AnyStateNode>,
+  stateNode: AnyStateNode = configuration[0].machine.root
 ): boolean {
-  if (stateNode.type === 'compound') {
-    return getChildren(stateNode).some(
-      (s) => s.type === 'final' && has(configuration, s)
-    );
-  }
-  if (stateNode.type === 'parallel') {
-    return getChildren(stateNode).every((sn) =>
-      isInFinalState(configuration, sn)
-    );
+  const { type } = stateNode;
+  if (['compound', 'parallel'].includes(type)) {
+    const children = getChildren(stateNode);
+    return type === 'compound'
+      ? children.some((sn) => sn.type === 'final' && configuration.includes(sn))
+      : children.every((sn) => isInFinalState(configuration, sn));
   }
```

Comment:
IMHO the previous version was more readable and was avoiding a temporary array and a redundant `.includes` check
```suggestion
  if (stateNode.type === 'compound') {
    return getChildren(stateNode).some(
      (s) => s.type === 'final' && configuration.includes(s)
    );
  }
  if (stateNode.type === 'parallel') {
    return getChildren(stateNode).every((sn) =>
      isInFinalState(configuration, sn)
    );
  }
```